torch/_inductor/codegen/cpp.py

```diff
@@ -914,8 +914,13 @@ def reduction(self, name, dtype, src_dtype, reduction_type, index, value):
                 self.reduction_prefix.writelines(
                     float16_reduction_prefix(reduction_type)
                 )
-            self.reduction_prefix.writeline(
-                f"{DTYPE_TO_CPP[dtype]} {tmpvar} = {reduction_init(reduction_type, dtype)};"
+            if dtype == torch.bfloat16:
+                self.reduction_prefix.writeline(
+                    f"{'float'} {tmpvar} = {reduction_init(reduction_type, dtype)};"
+                )
+            else:
+                self.reduction_prefix.writeline(
+                    f"{DTYPE_TO_CPP[dtype]} {tmpvar} = {reduction_init(reduction_type, dtype)};"
```

Comment:
how about torch.float16?

Reply:
BF16/FP16 can be unified. Removed the specific FP16 path for reduction init and removed 3 expected failure UT for FP16.